integration_test/test/signin_test.dart

```diff
@@ -0,0 +1,102 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:flutter_template/core/di/di_provider.dart';
+import 'package:flutter_template/core/model/authentication_status.dart';
+import 'package:flutter_template/main.dart' as app;
+import 'package:flutter_template/ui/extensions/context_extensions.dart';
+import 'package:flutter_template/ui/router/app_router.dart';
+import 'package:flutter_template/ui/section/error_handler/global_event_handler_cubit.dart';
+import 'package:flutter_template/ui/signin/signin_screen.dart';
+import 'package:flutter_template/ui/welcome/welcome_screen.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:rxdart/rxdart.dart';
+
+import '../common/general_helpers.dart';
+import '../common/repository_mocks.dart';
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  late MockSessionRepository mockSessionRepository;
+  late AppRouter appRouter;
+  late BuildContext context;
+  late StreamController<AuthenticationStatus> statusController;
+
+  setUp(() async {
+    mockSessionRepository = MockSessionRepository();
+    statusController = BehaviorSubject()
+      ..add(AuthenticationStatus.unauthenticated);
+    when(() => mockSessionRepository.status)
+        .thenAnswer((_) => statusController.stream);
+    appRouter = AppRouter(mockSessionRepository);
+    await commonSetup(mockSessionRepository, appRouter);
+  });
+
+  tearDown(() {
+    DiProvider.instance.reset();
+    statusController.close();
+  });
+  group('SignIn screen tests', () {
```

Comment:
can we extract this into a method?

Reply:
I think this case isn't long enough but if you are testing a form you might end up with a very long method main that test everything

Reply:
Do you mean to divide each group into different methods?

Reply:
yes